design-proposals/storage-class-migration/storage-class-migration.md

```diff
@@ -0,0 +1,276 @@
+# Overview
+This design proposal extends KubeVirt APIs to enable the migration between PVCs or storage classes.
+
+## Motivation
+In some circumstances, the guest owner may want to have the option of switching the storage provider or class for a group of volumes used by an active VM. The VM must currently be turned off and recreated for this operation causing service disruptions and downtime.
+
+
+## Goals
+Design and develop an API to handle storage migrations.
+
+ ## Non  Goals
+*  This features is not directly aimed for live migrating the VM using RWO
+   volumes. Hence, the API covers and refers only to the storage migration.
+
+ ## Definition of Users
+ This feature addresses for all KubeVirt users who want to move to another storage classes or PVCs without workload interuptions.
+
+## User Stories
+  1. A storage class of volumes used by a running VM is deprecated
+  2. A more performant storage class is now available in the cluster and the user wants start using it.
+  3. The storage device is filling up
+
+## Repos
+
+  - [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+This proposal introduces a new CRD `StorageMigration`
+
+```golang
+type StorageMigration struct {
+        metav1.TypeMeta   `json:",inline"`
+        metav1.ObjectMeta `json:"metadata,omitempty"`
+        Spec StorageMigrationSpec `json:"spec"`
+        Status *StorageMigrationStatus `json:"status,omitempty"`
+}
+
+// StorageMigrationList is a list of StorageMigration resources
+type StorageMigrationList struct {
+        metav1.TypeMeta `json:",inline"`
+        metav1.ListMeta `json:"metadata"`
+        Items []StorageMigration `json:"items"`
+}
+
+type MigratedVolume struct {
+        SourcePvc      string `json:"sourcePvc,omitempty" valid:"required"`
```

Comment:
IIRC you either have `omitempty` indicating its optional, or `valid:"required"` indicating the field is required. You can't combine the two.